### ISSUE TITLE
Using --disable-cache prevent to pull with a custom file name

### ISSUE THREAD

**Issue by sapetnioc** [apptainer/singularity#5628]
### Version of Singularity:

What version of Singularity are you using? Run:

```
$ singularity version
singularity version 3.6.4-1
```

The bug is only on rather recent versions

### Expected behavior

What did you expect to see when you do...?

```
singularity pull --disable-cache alpine.sif docker://alpine
```

I expect to see a file `alpine.sif` in the current directory.

### Actual behavior

What actually happend? Why was it incorrect?

The generated image is in `/tmp/sbuild-tmp-cache-*` where `*` is a random value.

### Steps to reproduce this behavior

How can others reproduce this issue/problem?

```
singularity pull --disable-cache alpine.sif docker://alpine
```

### What OS/distro are you running

```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic


```

### How did you install Singularity

From sources

**Comment by dcsouthwick** [comment]
Same issue here - affected versions:
*  Singularity 3.6.4-1.fc33 as distributed on system/update-testing
* Singularity 3.6.3-1-el7 as distributed by EPEL7

From 3.5.3 this works as expected

Further, the destination directory flag is also ignored:
`singularity pull --disable-cache --dir /tmp/images/ alpine.sif docker://alpine`  results in the same unknown sandbox folder in /tmp. 

(`--sandbox` was _never_ called)

**Comment by dtrudg** [comment]
~Looks to be a regression from the tmpdir security fixes~. Will work on this asap.

**Comment by dtrudg** [comment]
> `singularity pull --disable-cache --dir /tmp/images/ alpine.sif docker://alpine` results in the same unknown sandbox folder in /tmp.

This may be a separate issue. A `pull` from a `docker` source is actually a `build` internally. The `--dir` flag is possibly not being honoured properly in this flow. Will break out a separate issue if this is confirmed.

**Comment by dtrudg** [comment]
> > `singularity pull --disable-cache --dir /tmp/images/ alpine.sif docker://alpine` results in the same unknown sandbox folder in /tmp.
> 
> This may be a separate issue. A `pull` from a `docker` source is actually a `build` internally. The `--dir` flag is possibly not being honoured properly in this flow. Will break out a separate issue if this is confirmed.

Initial thought was incorrect - this issue has the same source, addressed in #5645
